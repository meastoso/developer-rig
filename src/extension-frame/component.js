import React, { Component } from 'react';
import PropTypes from 'prop-types';
import './component.sass';
const { ExtensionPlatform } = window['extension-coordinator'];

export const EXT_FRAME_WRAPPER_CLASS = 'extension-frame-wrapper';

export class ExtensionFrame extends Component {
  constructor() {
    super(...arguments);

    this._boundOnFrameDoubleClick = this._onFrameDoubleClick.bind(this);
    this._boundOnIdentityLinked = this._onIdentityLinked.bind(this);
    this._boundOnModalRequested = this._onModalRequested.bind(this);
    this._boundIframeHostRefHandler = this._iframeHostRefHandler.bind(this);

  }

  componentDidMount() {
    const { ExtensionFrame } = window['extension-coordinator'];
    this.extensionFrame = new ExtensionFrame({
      anchor: this.props.type,
      channelId: this.props.extension.channelId,
      loginId: null,
      dobbin: {
        trackEvent: (eventName, properties, services) => { },
      },
      extension: this.props.extension,
      iframeClassName: this.props.iframeClassName,
      mode: this.props.mode,
      platform: ExtensionPlatform.Web,
      parentElement: this.parentElement,
      trackingProperties: {},
    });

    if (this.parentElement) {
      this.parentElement.addEventListener('dblclick', this._boundOnFrameDoubleClick);
    }
    this.extensionFrame.on('identityLinked', this._boundOnIdentityLinked);
    this.extensionFrame.on('requestModal', this._boundOnModalRequested);
  }

  render() {
    return (
      <div
        ref={this._boundIframeHostRefHandler}
        className={EXT_FRAME_WRAPPER_CLASS}
      />
    );
  }

  _onIdentityLinked(isLinked) {
    const { extension, onIdentityLinked } = this.props;
    if (onIdentityLinked) {
      onIdentityLinked(extension.id, isLinked);
    }
  }

  _onFrameDoubleClick(evt) {
    evt.preventDefault();
  }

  _onModalRequested(confirmationRequest){}

  _iframeHostRefHandler(element) {
    this.parentElement = element;
  }
}

ExtensionFrame.propTypes = {
  iframeClassName: PropTypes.string.isRequired,
  extension: PropTypes.object.isRequired,
  type: PropTypes.string.isRequired,
  mode: PropTypes.string.isRequired,
};
