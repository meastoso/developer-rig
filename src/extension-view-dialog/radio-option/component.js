import React, { Component } from 'react';
import PropTypes from 'prop-types';
import './component.sass';

export class RadioOption extends Component {
  render() {
    return (
      <div>
        <label className="option-label">
            <input type="radio" name={this.props.name} value={this.props.value} onChange={this.props.onChange} defaultChecked={this.props.checked}/>
            {this.props.value}
        </label>
      </div>
    )
  }
}

RadioOption.propTypes = {
  name: PropTypes.string,
  value: PropTypes.string,
  onChange: PropTypes.func,
  checked: PropTypes.bool
}
