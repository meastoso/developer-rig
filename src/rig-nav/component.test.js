import { setupShallowTest } from '../tests/enzyme-util/shallow';
import { RigNav } from './component';
import { EXTENSION_VIEWS } from '../constants/nav-items';

describe('<RigNav />', () => {
  const setupShallow = setupShallowTest(RigNav, () => ({
    openConfigurationsHandler: jest.fn(),
    viewerHandler: jest.fn(),
    configHandler: jest.fn(),
    liveConfigHandler: jest.fn(),
    selectedView: 'test',
    error: '',
  }));

  it('renders correctly', () => {
    const { wrapper } = setupShallow();
    expect(wrapper).toMatchSnapshot();
  });

  it('renders an error', () => {
    const { wrapper } = setupShallow({
      error: 'test error',
    });
    expect(wrapper.find('.top-nav-error').text().trim()).toBe('test error');
  });

  it('correctly handles clicks on each tab', () => {
    const { wrapper } = setupShallow();
    wrapper.find('a.top-nav-item').forEach(tab => {
      tab.simulate('click');
    });
    expect(wrapper.instance().props.viewerHandler).toHaveBeenCalled();
    expect(wrapper.instance().props.configHandler).toHaveBeenCalled();
    expect(wrapper.instance().props.liveConfigHandler).toHaveBeenCalled();
    expect(wrapper.instance().props.openConfigurationsHandler).toHaveBeenCalled();
  });
});
