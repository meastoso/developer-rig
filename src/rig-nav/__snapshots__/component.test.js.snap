// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<RigNav /> renders correctly 1`] = `
ShallowWrapper {
  "length": 1,
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <RigNav
    configHandler={[Function]}
    error=""
    liveConfigHandler={[Function]}
    openConfigurationsHandler={[Function]}
    selectedView="test"
    viewerHandler={[Function]}
/>,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "children": Array [
        <a
          className="top-nav-item"
          onClick={[Function]}
>
          Extension Views
</a>,
        <a
          className="top-nav-item"
          onClick={[Function]}
>
          Broadcaster Config
</a>,
        <a
          className="top-nav-item"
          onClick={[Function]}
>
          Live Config
</a>,
        <a
          className="top-nav-item"
          onClick={[Function]}
>
          Configurations
</a>,
      ],
      "className": "top-nav",
    },
    "ref": null,
    "rendered": Array [
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": "Extension Views",
          "className": "top-nav-item",
          "onClick": [Function],
        },
        "ref": null,
        "rendered": "Extension Views",
        "type": "a",
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": "Broadcaster Config",
          "className": "top-nav-item",
          "onClick": [Function],
        },
        "ref": null,
        "rendered": "Broadcaster Config",
        "type": "a",
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": "Live Config",
          "className": "top-nav-item",
          "onClick": [Function],
        },
        "ref": null,
        "rendered": "Live Config",
        "type": "a",
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": "Configurations",
          "className": "top-nav-item",
          "onClick": [Function],
        },
        "ref": null,
        "rendered": "Configurations",
        "type": "a",
      },
    ],
    "type": "div",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": Array [
          <a
            className="top-nav-item"
            onClick={[Function]}
>
            Extension Views
</a>,
          <a
            className="top-nav-item"
            onClick={[Function]}
>
            Broadcaster Config
</a>,
          <a
            className="top-nav-item"
            onClick={[Function]}
>
            Live Config
</a>,
          <a
            className="top-nav-item"
            onClick={[Function]}
>
            Configurations
</a>,
        ],
        "className": "top-nav",
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": "Extension Views",
            "className": "top-nav-item",
            "onClick": [Function],
          },
          "ref": null,
          "rendered": "Extension Views",
          "type": "a",
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": "Broadcaster Config",
            "className": "top-nav-item",
            "onClick": [Function],
          },
          "ref": null,
          "rendered": "Broadcaster Config",
          "type": "a",
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": "Live Config",
            "className": "top-nav-item",
            "onClick": [Function],
          },
          "ref": null,
          "rendered": "Live Config",
          "type": "a",
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": "Configurations",
            "className": "top-nav-item",
            "onClick": [Function],
          },
          "ref": null,
          "rendered": "Configurations",
          "type": "a",
        },
      ],
      "type": "div",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
      },
    },
  },
}
`;
